scipy/sparse/_bsr.py

```diff
@@ -729,11 +628,211 @@ def isspmatrix_bsr(x):
 
 # This namespace class separates array from matrix with isinstance
 class bsr_array(_bsr_base, sparray):
-    pass
+    """
+    Block Sparse Row format sparse array.
+
+    This can be instantiated in several ways:
+        bsr_array(D, [blocksize=(R,C)])
+            where D is a dense matrix or 2-D ndarray.
```

Comment:
I think dense matrix refers to `np.matrix` which have been deprecated. Hence I think that we better remove this mention here.
```suggestion
            where D is a 2-D ndarray.
```

Reply:
Sounds good!